mitiq/shadows/shadows.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+"""Classical shadow estimation for quantum circuits. Based on the paper"""
+
+from typing import Optional, Callable, List, Dict, Any
+
+import cirq
+import numpy as np
+from numpy.typing import NDArray
+
+import mitiq
+from mitiq import MeasurementResult
+from mitiq.shadows import (
+    random_pauli_measurement,
+    shadow_state_reconstruction,
+    expectation_estimation_shadow,
+)
+from mitiq.shadows.shadows_utils import (
+    n_measurements_tomography_bound,
+    n_measurements_opts_expectation_bound,
+)
+
+
+def execute_with_shadows(
```

Comment:
All `execute_with_xxx` functions in Mitiq have a similar signature, they take as input a `circuit`, an `executor`, an optional single `mitiq.Observable` (a linear combination of `PauliString`s) and they produce as output a single expectation value. The motivation for such a fixed structure is to easily switch between different error mitigation techniques, or to easily combine multiple techniques.

This function instead takes as input a list of many `mitiq.PauliString` objects, corresponding to many Pauli "observables", and returns a dictionary with all the results.

I agree that, due to the structure of the classical shadow algorithm, having many Pauli strings and therefore many expectation values is essential. However I wonder if we should use a different name for this function. E.g. something like `run_shadow_protocol()`.

[**Optional**]
If you think it can be useful, we could also add a new function `execute_with_shadows()`, which takes as input a single `mitiq.Observable` (which is a linear combination of many pauli strings) and evaluates its single expectation value by internally calling the general `run_shadow_protocol()` function and by taking a linear combination of all the individual expectation values associated to each pauli string.
I don't know if having this new function is important at this stage. It could become more useful when we'll have robust shadows (with error mitigation).